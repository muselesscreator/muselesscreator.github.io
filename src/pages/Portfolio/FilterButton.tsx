import classNames from 'classnames';

type FilterButtonProps = {
  isActive: boolean;
  filter: string;
  handleFilterChange: (category: string) => () => void;
};
const FilterButton = ({
  isActive,
  filter,
  handleFilterChange,
}: FilterButtonProps) => {
  return (
    <li className="filter-item">
      <button
        className={classNames({ active: isActive })}
        onClick={handleFilterChange(filter)}
      >
        {filter}
      </button>
    </li>
  );
};
export default FilterButton;
