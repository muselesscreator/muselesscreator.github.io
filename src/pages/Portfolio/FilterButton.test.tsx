import {
  vi,
  describe,
  it,
  test,
  expect,
} from 'vitest';
import { shallow, setMockName } from '@muselesscreator/react-shallow-snapshot';

import FilterButton from './FilterButton';

const onClick = vi.fn();
const props = {
  isActive: true,
  handleFilterChange: (filter: string) => {
    setMockName(onClick, `handleFilterChange(${filter})`);
    return onClick as () => void;
  },
  filter: 'Test Filter',
};

setMockName(props.handleFilterChange, 'props.onClick');

describe('FilterButton component', () => {
  test('active snapshot', () => {
    const el = shallow(<FilterButton {...props} />);
    expect(el.snapshot).toMatchSnapshot();
    const button = el.instance.findByType('button')[0];
    expect(button.props.className).toEqual('active');
    expect(button.props.onClick).toEqual(props.handleFilterChange(props.filter));
  });
  test('inactive snapshot', () => {
    const el = shallow(<FilterButton {...props} isActive={false} />);
    expect(el.snapshot).toMatchSnapshot();
    const button = el.instance.findByType('button')[0];
    expect(button.props.className).toEqual('');
    expect(button.props.onClick).toEqual(props.handleFilterChange(props.filter));
  });
});

