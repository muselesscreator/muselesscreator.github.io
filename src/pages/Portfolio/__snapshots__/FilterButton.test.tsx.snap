// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterButton component > active snapshot 1`] = `
"<li
  className="filter-item"
>
  <button
    className="active"
    onClick={[Function handleFilterChange(Test Filter)]}
  >
    Test Filter
  </button>
</li>"
`;

exports[`FilterButton component > inactive snapshot 1`] = `
"<li
  className="filter-item"
>
  <button
    className=""
    onClick={[Function handleFilterChange(Test Filter)]}
  >
    Test Filter
  </button>
</li>"
`;
